core/src/main/java/com/sequenceiq/cloudbreak/service/stackpatch/UnboundRestartPatchService.java

```diff
@@ -0,0 +1,107 @@
+package com.sequenceiq.cloudbreak.service.stackpatch;
+
+import static com.sequenceiq.cloudbreak.domain.stack.StackPatchType.UNBOUND_RESTART;
+
+import java.util.Set;
+import java.util.concurrent.TimeUnit;
+
+import javax.inject.Inject;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Service;
+
+import com.dyngr.Polling;
+import com.dyngr.core.AttemptResult;
+import com.dyngr.core.AttemptResults;
+import com.sequenceiq.cloudbreak.auth.ThreadBasedUserCrnProvider;
+import com.sequenceiq.cloudbreak.cloud.model.Image;
+import com.sequenceiq.cloudbreak.core.CloudbreakImageNotFoundException;
+import com.sequenceiq.cloudbreak.domain.stack.Stack;
+import com.sequenceiq.cloudbreak.domain.stack.StackPatchType;
+import com.sequenceiq.cloudbreak.domain.stack.instance.InstanceMetaData;
+import com.sequenceiq.cloudbreak.service.cluster.flow.ClusterOperationService;
+import com.sequenceiq.cloudbreak.service.stack.StackImageService;
+import com.sequenceiq.flow.api.model.FlowCheckResponse;
+import com.sequenceiq.flow.api.model.FlowIdentifier;
+import com.sequenceiq.flow.service.FlowService;
+
+@Service
+public class UnboundRestartPatchService extends ExistingStackPatchService {
+
+    private static final String AFFECTED_STACK_VERSION = "7.2.11";
+
+    private static final Set<String> AFFECTED_IMAGE_IDS = Set.of(
+            "26cd5a65-cd5c-457d-8d48-9caf1a486516",
+            "19cf97b8-56d8-4be9-b317-998eea99d884",
+            "c24acec3-9110-4474-9082-3620deac0910"
+    );
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(UnboundRestartPatchService.class);
+
+    @Inject
+    private StackImageService stackImageService;
+
+    @Inject
+    private ClusterOperationService clusterOperationService;
+
+    @Inject
+    private FlowService flowService;
+
+    @Override
+    public StackPatchType getStackFixType() {
+        return UNBOUND_RESTART;
+    }
+
+    @Override
+    public boolean isAffected(Stack stack) {
+        try {
+            boolean affected = false;
+            if (AFFECTED_STACK_VERSION.equals(stack.getStackVersion())) {
+                Image image = stackImageService.getCurrentImage(stack);
+                affected = AFFECTED_IMAGE_IDS.contains(image.getImageId());
+                LOGGER.debug("Stack {} with version {} and image {} is {} by unbound service restart bug",
+                        stack.getResourceCrn(), stack.getStackVersion(), image.getImageId(), affected ? "affected" : "not affected");
+            } else {
+                LOGGER.debug("Stack {} with version {} is not affected by unbound service restart bug", stack.getResourceCrn(), stack.getStackVersion());
+            }
+            return affected;
+        } catch (CloudbreakImageNotFoundException e) {
+            LOGGER.warn("Image not found for stack " + stack.getResourceCrn(), e);
+            throw new IllegalStateException("Image not found for stack " + stack.getResourceCrn(), e);
```

Comment:
is there any specific reason to throw this `IllegalStateException` here?

Reply:
CloudbreakRuntimeException?

Reply:
Changed to CloudbreakRuntimeException